catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -77,20 +80,75 @@ public ResultIterable(CatalogFramework catalogFramework, QueryRequest queryReque
      * @param queryRequest  request used to retrieve the results.
      */
     public ResultIterable(QueryFunction queryFunction, QueryRequest queryRequest) {
+        this(queryFunction, queryRequest, 0);
+    }
+
+    /**
+     * Creates an iterable that will call the {@link CatalogFramework} to retrieve the results
+     * that match the {@link QueryRequest} provided.
+     *
+     * @param catalogFramework reference to the {@link CatalogFramework} to call to retrieve the
+     *                         results.
+     * @param queryRequest     request used to retrieve the results.
+     * @param maxResultCount   maximum number of results in total to query for
+     */
+    public ResultIterable(CatalogFramework catalogFramework, QueryRequest queryRequest,
+            int maxResultCount) {
+        notNull(catalogFramework, "CatalogFramework cannot be null");
+        notNull(queryRequest, "Query request cannot be null");
+        Validate.isTrue(maxResultCount >= 0,
+                "Max Results must be zero or positive",
+                maxResultCount);
+
+        this.queryFunction = catalogFramework::query;
+        this.queryRequest = queryRequest;
+        this.maxResultCount = maxResultCount;
+    }
+
+    /**
+     * Creates an iterable that will call a {@link QueryFunction} to retrieve the results
+     * that match the {@link QueryRequest} provided.
+     *
+     * @param queryFunction  reference to the {@link QueryFunction} to call to retrieve the
+     *                       results.
+     * @param queryRequest   request used to retrieve the results.
+     * @param maxResultCount maximum number of results in total to query for
+     */
+    public ResultIterable(QueryFunction queryFunction, QueryRequest queryRequest,
+            int maxResultCount) {
         notNull(queryFunction, "Query function cannot be null");
         notNull(queryRequest, "Query request cannot be null");
+        Validate.isTrue(maxResultCount >= 0,
+                "Max Results must be zero or positive",
+                maxResultCount);
 
         this.queryFunction = queryFunction;
         this.queryRequest = queryRequest;
+        this.maxResultCount = maxResultCount;
     }
 
     @Override
     public Iterator<Result> iterator() {
+        if (maxResultCount > 0) {
+            return limit(new QueryResultIterator(queryFunction, queryRequest), maxResultCount);
+        }
         return new QueryResultIterator(queryFunction, queryRequest);
     }
 
+    /**
```

Comment:
❗️ If we decide to keep this method, we should provide a description for this method.